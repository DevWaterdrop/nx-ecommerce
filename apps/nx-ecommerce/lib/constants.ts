export const LOVELY_ERROR = 'I HATE TYPESCRIPT+STRAPI+GRAPHQL';
export const MAXIMUM_IN_CART = 16;

export enum PRICE_FILTER {
  ZERO_TWENTY = 'zt',
  TWENTY_FORTY = 'tf',
  FORTY_EIGHTY = 'fe',
  EIGHTY_PLUS = 'ep',
}

export enum SORT_FILTER {
  PRICE_LOW_HIGH = 'plh',
  PRICE_HIGH_LOW = 'phl',
  NAME = 'name',
}

export const PRICES = [
  { value: PRICE_FILTER.ZERO_TWENTY, label: '0.00$ - 19.99$' },
  { value: PRICE_FILTER.TWENTY_FORTY, label: '20.00$ - 39.99$' },
  { value: PRICE_FILTER.FORTY_EIGHTY, label: '40.00$ - 79.99$' },
  { value: PRICE_FILTER.EIGHTY_PLUS, label: '80.00$+' },
];

export const SORT = [
  { value: SORT_FILTER.PRICE_LOW_HIGH, label: 'Price: low to high' },
  { value: SORT_FILTER.PRICE_HIGH_LOW, label: 'Price: high to low' },
  { value: SORT_FILTER.NAME, label: 'Name' },
];
