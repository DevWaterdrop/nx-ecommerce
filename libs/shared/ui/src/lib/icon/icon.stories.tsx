import { Story, Meta } from '@storybook/react';
import { Icon, IconProps } from './icon';

export default {
  component: Icon,
  title: 'Components/Icon',
} as Meta;

const Template: Story<IconProps> = (args) => <Icon {...args} />;

export const Primary = Template.bind({});
Primary.args = {};
