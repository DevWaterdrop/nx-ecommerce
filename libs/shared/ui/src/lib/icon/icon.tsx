import shoppingCartSVG from '../../assets/icons/shopping-cart.svg';
import heartSVG from '../../assets/icons/heart.svg';
import heartFilledSVG from '../../assets/icons/heart-filled.svg';
import userSVG from '../../assets/icons/user.svg';
import shoppingCartBlackSVG from '../../assets/icons/shopping-cart-black.svg';
import arrowRightSVG from '../../assets/icons/arrow-right.svg';
import arrowRightBlackSVG from '../../assets/icons/arrow-right-black.svg';
import Image from 'next/image';
import clsx from 'clsx';
import { Spinner } from './local-components/spinner';

export interface IconProps {
  icon?: keyof typeof ICONS;
  loading?: boolean;
  classes?: string;
  size?: typeof SIZES[number];
}

export const ICONS = {
  cart: shoppingCartSVG,
  cartBlack: shoppingCartBlackSVG,
  heart: heartSVG,
  heartFilled: heartFilledSVG,
  user: userSVG,
  arrowRight: arrowRightSVG,
  arrowRightBlack: arrowRightBlackSVG,
} as const;

const SIZES = ['sm', 'base', 'lg'] as const;

export const Icon: React.FC<IconProps> = (props: IconProps) => {
  const { icon = 'cart', classes, size = 'base', loading } = props;

  return (
    <div
      className={clsx(
        'relative',
        'flex items-center',
        'select-none pointer-events-none',
        // SIZES
        size === 'sm' && 'h-4 w-4  sm:(h-6 w-6)',
        size === 'base' && 'h-5 w-5',
        size === 'lg' && 'h-8 w-8',
        //
        classes
      )}
      aria-hidden="true"
    >
      {loading ? <Spinner /> : <Image src={ICONS[icon]} layout="fill" alt="" />}
    </div>
  );
};
