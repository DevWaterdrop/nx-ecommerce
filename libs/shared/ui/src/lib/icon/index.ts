import { Icon, IconProps, ICONS } from './icon';

export { Icon, ICONS };

export type { IconProps };
