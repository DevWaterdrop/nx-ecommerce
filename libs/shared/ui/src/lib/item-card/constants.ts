import { ItemCardProps } from './item-card';
import flowerJPEG from '../../props/flower.jpeg';
import roomJPEG from '../../props/room.jpeg';

const ITEM: Required<ItemCardProps['item']> = {
  price: 65,
  name: 'Lovely flower',
  id: '#',
  images: [flowerJPEG, roomJPEG],
  smallDescription: 'For you, for me, for us',
  tag: 'new',
};

export const DEFAULT_PROPS: ItemCardProps = {
  item: ITEM,
  size: 'base',
  handleClick: () => console.log('clicked'),
};
