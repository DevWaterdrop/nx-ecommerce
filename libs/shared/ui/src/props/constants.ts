import { ButtonProps } from '../lib/button';
import { CartItemProps } from '../lib/cart-item';
import { CartSidebarProps } from '../lib/cart-sidebar';
import { CategoryCardProps } from '../lib/category-card';
import { IconProps } from '../lib/icon/icon';
import { ItemCardProps } from '../lib/item-card';
import { NavButtonProps } from '../lib/nav-button';
import { OrdersTableProps } from '../lib/orders-table';
import { ProductGalleryProps } from '../lib/product-gallery';
import { ProductInformationProps } from '../lib/product-information';
import { ProductSidebarProps } from '../lib/product-sidebar';
import { SearchbarProps } from '../lib/searchbar';
import { SearchbarItemProps } from '../lib/searchbar-item';
import { SelectProps } from '../lib/select';
import flowerJPEG from './flower.jpeg';
import roomJPEG from './room.jpeg';

export const ITEM: ItemCardProps['item'] = {
  slug: 'lovely-flower',
  price: 65,
  name: 'Lovely flower',
  images: {
    data: [
      {
        attributes: {
          url: flowerJPEG,
        },
      },
      {
        attributes: {
          url: roomJPEG,
        },
      },
    ],
  },
  smallDescription: 'For you, for me, for us',
  description: 'Lovely description',
};

export const ITEM_SECOND: ItemCardProps['item'] = {
  slug: 'roomy',
  price: 50,
  name: 'Roomy',
  images: {
    data: [
      {
        attributes: {
          url: flowerJPEG,
        },
      },
      {
        attributes: {
          url: roomJPEG,
        },
      },
    ],
  },
  smallDescription: 'Hmmm, nice vase',
  description: 'Vaaaase',
};

export const ITEM_CARD_DEFAULT_PROPS: ItemCardProps = {
  item: ITEM,
  size: 'base',
  handleCartClick: () => console.log('cart clicked'),
  handleAmountCartClick: () => console.log('amount cart clicked'),
  handleFavoriteClick: () => console.log('favorite clicked'),
};

export const CATEGORY_CARD_DEFAULT_PROPS: CategoryCardProps = {
  category: {
    smallDescription:
      'Lorem ipsum dolor sit amet, consectetur adipisicing elit.',
    image: {
      data: {
        attributes: {
          url: roomJPEG,
        },
      },
    },
    slug: '1',
  },
};

export const NAV_BUTTON_DEFAULT_PROPS: NavButtonProps = {
  type: 'heart',
};

export const ICON_DEFAULT_PROPS: IconProps = {
  icon: 'cart',
};

export const BUTTON_DEFAULT_PROPS: ButtonProps = {
  children: 'hello',
  tag: 'button',
  size: 'base',
  type: 'primary',
  round: 'full',
};

export const SEARCHBAR_ITEM: SearchbarProps['items'][number] = {
  id: '111',
  attributes: {
    slug: 'lovely-flower',
    smallDescription: 'For you, for me, for us',
    name: 'Lovely flower',
    images: {
      data: [
        {
          attributes: {
            url: flowerJPEG,
          },
        },
      ],
    },
  },
};

export const SEARCHBAR_ITEM_DEFAULT_PROPS: SearchbarItemProps = {
  item: {
    slug: 'lovely-flower',
    smallDescription: 'For you, for me, for us',
    name: 'Lovely flower',
    images: {
      data: [
        {
          attributes: {
            url: flowerJPEG,
          },
        },
      ],
    },
  },
  handleClick: () => void 1,
};

export const SELECT_ITEM = {
  label: 'Label1',
  value: 'value1',
};

export const SELECT_DEFAULT_PROPS: Required<
  Omit<SelectProps, 'placeholder' | 'multiple'>
> = {
  items: [SELECT_ITEM],
  initial: ['value1'],
  onChange: () => console.log('changed'),
};

export const PRODUCT: ProductSidebarProps['product'] = {
  slug: 'lovely-flower',
  price: 65,
  name: 'Lovely flower',
  images: {
    data: [
      {
        attributes: {
          url: flowerJPEG,
        },
      },
      {
        attributes: {
          url: roomJPEG,
        },
      },
    ],
  },
  smallDescription: 'For you, for me, for us',
  description: 'Lovely description',
  information: [
    {
      title: 'Title1',
      content: 'content1',
    },
  ],
};

export const PRODUCT_SIDEBAR_DEFAULT_PROPS: ProductSidebarProps = {
  product: PRODUCT,
  handleCartClick: () => {
    console.log('cart');
  },
  handleRemoveCartClick: () => {
    console.log('remove cart');
  },
  handleFavoriteClick: () => {
    console.log('favorite');
  },
  cartValue: new Map([]),
  favoritesValue: new Set('lovely-flower'),
};

export const PRODUCT_INFORMATION_DEFAULT_PROPS: ProductInformationProps = {
  information: [...Array(10).keys()].map((_value, index) => ({
    title: `Title${index + 1}`,
    content: `content${index + 1}`,
  })),
};

export const PRODUCT_GALLERY_DEFAULT_PROPS: ProductGalleryProps = {
  images: [
    {
      attributes: {
        url: flowerJPEG,
      },
    },
    {
      attributes: {
        url: roomJPEG,
      },
    },
  ],
};

const PRODUCT_ORDER_TABLE: OrdersTableProps['orders'][number]['items'][number]['product'] =
  {
    name: 'Flower',
    slug: 'slug-flower',
    description: '',
    smallDescription: '',
    price: 50,
  };

export const ORDERS_TABLE_DEFAULT_PROPS: OrdersTableProps = {
  orders: [
    {
      id: 'id-1',
      amount: 50,
      items: [{ id: '1', amount: 1, product: PRODUCT_ORDER_TABLE }],
    },
    {
      id: 'id-2',
      amount: 100,
      items: [{ id: '2', amount: 2, product: PRODUCT_ORDER_TABLE }],
    },
  ],
};

export const CART_SIDEBAR_DEFAULT_PROPS: CartSidebarProps = {
  cartTotal: 100,
  isUser: true,
  handleCheckoutClick: async () => {
    console.log('checkout');
  },
  isLoading: false,
};

export const CART_ITEM_DEFAULT_PROPS: CartItemProps = {
  inCartAmount: 1,
  item: PRODUCT,
  handleCartMinusClick: () => {
    console.log('cart minus');
  },
  handleCartPlusClick: () => {
    console.log('cart plus');
  },
  handleRemoveFromCart: () => {
    console.log('cart remove');
  },
  toggleFavorite: () => {
    console.log('favorite');
  },
};
