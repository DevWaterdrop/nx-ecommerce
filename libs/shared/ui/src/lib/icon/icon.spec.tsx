import { ICON_DEFAULT_PROPS } from '../../props/constants';
import { render, RenderResult, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import { Icon } from './icon';

describe('Icon', () => {
  let renderResult: RenderResult;
  const props = ICON_DEFAULT_PROPS;

  beforeEach(() => {
    renderResult = render(<Icon {...props} />);
  });

  it('should render successfully', () => {
    const { baseElement } = renderResult;

    const tree = renderer.create(<Icon {...props} />).toJSON();
    expect(tree).toMatchSnapshot();

    expect(baseElement).toBeTruthy();
  });

  it('when loading show spinner instead of icon', () => {
    expect(screen.queryByTestId('spinner')).not.toBeInTheDocument();

    renderResult.rerender(<Icon {...props} loading={true} />);

    expect(screen.queryByTestId('spinner')).toBeInTheDocument();
  });
});
