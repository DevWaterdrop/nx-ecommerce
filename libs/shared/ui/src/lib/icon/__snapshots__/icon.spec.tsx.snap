// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render successfully 1`] = `
<div
  aria-hidden="true"
  className="relative flex items-center select-none pointer-events-none h-5 w-5"
>
  Next image stub
</div>
`;
